components/automate-ui/src/app/modules/team/team-details/team-details.component.ts

```diff
@@ -214,9 +191,9 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
   saveTeam(): void {
     this.saveSuccessful = false;
     this.saveInProgress = true;
-    this.updateNameForm.controls['name'].disable();
-    const name: string = this.updateNameForm.controls.name.value.trim();
-    const projects = Object.keys(this.projects).filter(id => this.projects[id].checked);
+    this.updateForm.controls['name'].disable();
+    const name: string = this.updateForm.controls.name.value.trim();
+    const projects: string[] = this.updateForm.controls.projects.value;
```

Comment:
Instead of using the `this.projects` intermediary and copying from there, we get the value directly from the form, which...